api/v1beta1/designate_webhook.go

```diff
@@ -340,72 +323,55 @@ func (spec *DesignateSpecCore) ValidateDesignateTopology(basePath *field.Path, n
 
 	// When a TopologyRef CR is referenced, fail if a different Namespace is
 	// referenced because is not supported
-	if spec.TopologyRef != nil {
-		if err := topologyv1.ValidateTopologyNamespace(spec.TopologyRef.Namespace, *basePath, namespace); err != nil {
-			allErrs = append(allErrs, err)
-		}
-	}
+	allErrs = append(allErrs, topologyv1.ValidateTopologyRef(
+		spec.TopologyRef, *basePath.Child("topologyRef"), namespace)...)
 
 	// When a TopologyRef CR is referenced with an override to DesignateAPI, fail
 	// if a different Namespace is referenced because not supported
-	if spec.DesignateAPI.TopologyRef != nil {
-		if err := topologyv1.ValidateTopologyNamespace(spec.DesignateAPI.TopologyRef.Namespace, *basePath, namespace); err != nil {
-			allErrs = append(allErrs, err)
-		}
-	}
+	apiPath := basePath.Child("designateAPI")
+	allErrs = append(allErrs,
+		spec.DesignateAPI.ValidateTopology(apiPath, namespace)...)
 
 	// When a TopologyRef CR is referenced with an override to DesignateBackendbind9
 	// fail if a different Namespace is referenced because not supported
-	if spec.DesignateBackendbind9.TopologyRef != nil {
-		if err := topologyv1.ValidateTopologyNamespace(spec.DesignateBackendbind9.TopologyRef.Namespace, *basePath, namespace); err != nil {
-			allErrs = append(allErrs, err)
-		}
-	}
+	bind9Path := basePath.Child("designateBackednBind9")
```

Comment:
designateBackednBind9 -> designateBackendBind9

Reply:
Done! (and thanks for catching it!)